### PR TITLE
Various small improvements

### PR DESCRIPTION
**Pull request by Luthaf**

### CODE REVIEW

File: include/chemfiles/Residue.hpp [hunk 1]
```diff
@@ -48,12 +48,7 @@ class CHFL_EXPORT Residue final {
         return name_;
     }
 
-    /// Get the residue identifier if it exists.
-    ///
-    /// @verbatim embed:rst:leading-slashes
-    /// This function returna an :cpp:class:`chemfiles::optional` value that is
-    /// close to C++17 ``std::optional``.
-    /// @endverbatim
+    /// Get the residue identifier, or `nullopt` if it does not exist
```
Comment (ezavod) [inline include/chemfiles/Residue.hpp]: Don't we use whole sentences every where else, so point at the end?

File: Doxyfile.in [hunk 2]
```diff
@@ -514,7 +551,7 @@ INTERNAL_DOCS          = NO
 # names in lower-case letters. If set to YES, upper-case letters are also
 # allowed. This is useful if you have classes or files whose names only differ
 # in case and if your file system supports case sensitive file names. Windows
-# and Mac users are advised to set this option to NO.
+# (including Cygwin) ands Mac users are advised to set this option to NO.
```
Comment (ezavod) [inline Doxyfile.in]: and~s~  :D

File: doc/src/classes/helpers.rst [hunk 3]
```diff
@@ -45,7 +45,7 @@ how to use these types.
 .. cpp:class:: template <class T> chemfiles::span
 
     A ``span<T>`` is a view inside a ``std::vector<T>`` providing all the
-    operations of a vector expect for memory allocation. The idea and
+    operations of a vector excepted for memory allocation. The idea and
```
Comment (ezavod) [inline doc/src/classes/helpers.rst]: As a non-native speaker hard to say, but does "... except for ..." sound better? So all allocations would be excepted (passive). =)

File: Doxyfile.in (hunk 2 above)
Comment (Luthaf) [reply Doxyfile.in]: that was automatically written by doygen, using `doxygen -u Doxyfile.in`, so the issue is on their side :smile:

File: doc/src/classes/helpers.rst (hunk 3 above)
Comment (Luthaf) [reply doc/src/classes/helpers.rst]: non-native speaker here too, but yes, this sounds nicer!

File: Doxyfile.in (hunk 2 above)
Comment (ezavod) [reply Doxyfile.in]: :joy: didn't thought about where it's coming from, but it's indeed in their [current master](https://github.com/doxygen/doxygen/blob/e8f268edd2e440c136735e878e4c9d79d57354ae/src/config.xml#L963). This just jumped at me while scrolling through.

**Pull request merged**